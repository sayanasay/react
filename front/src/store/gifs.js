import {
  GET_GIFS_SUCCESS,
  GET_GIFS_REQUEST,
  GET_GIFS_FAILURE,
  SET_SEARCH_TEXT,
} from "./types/gifsTypes";

export const STATUSES = {
  IDLE: 0,
  REQUEST: 1,
  SUCCESS: 2,
  FAILURE: 3,
};

const initialState = {
  gifsList: [],
  searchText: "",
  request: STATUSES.IDLE,
  error: null,
};

const gifs = (state = initialState, { type, payload }) => {
  switch (type) {
    case GET_GIFS_REQUEST:
      return { ...state, request: STATUSES.REQUEST };

    case GET_GIFS_SUCCESS:
      return {
        ...state,
        gifsList: payload,
        request: STATUSES.SUCCESS,
        error: null,
      };

    case GET_GIFS_FAILURE:
      return { ...state, request: STATUSES.FAILURE, error: payload };

    case SET_SEARCH_TEXT:
      return { ...state, searchText: payload };

    default:
      return state;
  }
};

export default gifs;