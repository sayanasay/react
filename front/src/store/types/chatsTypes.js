const SET_CHATS = 'SET_CHATS';

export { SET_CHATS };