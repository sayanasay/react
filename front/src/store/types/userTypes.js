const USER_LOGIN = 'USER_LOGIN';
const SET_ERROR = 'SET_ERROR';
const USER_LOGOUT = 'USER_LOGOUT'

export { USER_LOGIN, SET_ERROR, USER_LOGOUT };