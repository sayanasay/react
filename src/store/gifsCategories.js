import {
  GET_CATEGORIES_REQUEST,
  GET_CATEGORIES_SUCCESS,
  GET_CATEGORIES_FAILURE,
} from "./types/gifsTypes";

export const STATUSES = {
  IDLE: 0,
  REQUEST: 1,
  SUCCESS: 2,
  FAILURE: 3,
};

const initialState = {
  categories: [],
  request: STATUSES.IDLE,
  error: null,
};

const gifsCategories = (state = initialState, { type, payload }) => {
  switch (type) {

    case GET_CATEGORIES_REQUEST:
      return { ...state, request: STATUSES.REQUEST };

    case GET_CATEGORIES_SUCCESS:
      return {
        ...state,
        categories: payload,
        request: STATUSES.SUCCESS,
        error: null,
      };

    case GET_CATEGORIES_FAILURE:
      return { ...state, request: STATUSES.FAILURE, error: payload };

    default:
      return state;
  }
};

export default gifsCategories;