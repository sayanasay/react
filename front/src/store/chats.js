import { SET_CHATS } from "./types/chatsTypes";

const initialState = [];

const chats = (state = initialState, { type, payload }) => {
  switch (type) {
    case SET_CHATS:
      return payload;

    default:
      return state;
  }
};

export default chats;
