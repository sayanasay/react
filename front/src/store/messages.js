import { SET_MESSAGES } from "./types/messagesTypes";

const initialState = [];

const messages = (state = initialState, { type, payload }) => {
  switch (type) {
    case SET_MESSAGES:
      return payload;

    default:
      return state;
  }
};

export default messages;