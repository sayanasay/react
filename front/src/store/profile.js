import { CHANGE_CHECKBOX } from "./types/profileTypes";

const initialState = {
  checkBox: false,
};

const profileReducer = (state = initialState, {type}) => {
  switch (type) {
    case CHANGE_CHECKBOX:
      return { checkBox: !state.checkBox };
    default:
      return state;
  }
}

export default profileReducer;