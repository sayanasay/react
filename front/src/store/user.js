import { SET_ERROR, USER_LOGIN, USER_LOGOUT } from "./types/userTypes";

const initialState =  { user: null, error: null };

const user = (state = initialState, { type, payload }) => {
  switch (type) {
    case USER_LOGIN:
      return { error: null, user: payload };

    case USER_LOGOUT:
      return { error: null, user: null };

    case SET_ERROR:
      return  { ...state, error: payload }

    default:
      return state;
  }
}

export default user;