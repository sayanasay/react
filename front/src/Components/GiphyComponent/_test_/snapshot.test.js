import "@testing-library/react";
import { initialState } from "../../../store/gifs";
import gifsCategories from "../../../store/gifsCategories";
import { GET_CATEGORIES_SUCCESS } from "../../../store/types/gifsTypes";

describe('Snapshot test', () => {
  it('returns correct state after GET_CATEGORIES_SUCCESS action', () => {
    const action = {
      type: GET_CATEGORIES_SUCCESS,
      payload: [1, 2, 3],
    }
    const gifCatStore = gifsCategories(initialState, action)
    expect(gifCatStore).toMatchSnapshot();
  });
})