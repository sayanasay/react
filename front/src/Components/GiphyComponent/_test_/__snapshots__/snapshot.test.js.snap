// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test returns correct state after GET_CATEGORIES_SUCCESS action 1`] = `
Object {
  "categories": Array [
    1,
    2,
    3,
  ],
  "error": null,
  "gifsList": Array [],
  "request": 2,
  "searchText": "",
}
`;
